californium-core/src/main/java/org/eclipse/californium/core/network/InMemoryMessageExchangeStore.java

```diff
@@ -192,18 +199,21 @@ public int assignMessageId(final Message message) {
 
 	private int registerWithMessageId(final Exchange exchange, final Message message) {
 
+		enableStatus = true;
 		int mid = message.getMID();
 		if (Message.NONE == mid) {
 			mid = assignMessageId(message);
 			if (Message.NONE != mid) {
 				KeyMID key = KeyMID.fromOutboundMessage(message);
 				if (exchangesByMID.putIfAbsent(key, exchange) != null) {
-					throw new IllegalArgumentException(String
-							.format("automatic message ID [%d] already in use, cannot register exchange", message.getMID()));
+					throw new IllegalArgumentException(String.format(
+							"automatic message ID [%d] already in use, cannot register exchange", message.getMID()));
 				}
+				LOGGER.debug("exchange added with generated mid {}, {}", key, message);
```

Comment:
this feels more like TRACE but I do not really know the context in which you need this info ...

Reply:
Searching for exchange leaks, I felt it was debugging :-),